draft-ietf-tls-cross-sni-resumption.md

```diff
@@ -116,6 +120,22 @@ merely lifts a performance-motivated "SHOULD NOT" recommendation from Section
 4.6.1.  Notably, it still relies on the client ensuring that the server
 certificate is valid for the new SNI at the time of session resumption.
 
+If the original server's assertion regarding supporting cross-name resumption
+turns out to be incorrect, the server receiving a misdirected ticket
+will not be able to decrypt it, and will therefore reject it.  This is secure,
+as session resumption may be safely rejected for any reason; however, such
+misconfiguration will waste tickets stored in the client's cache, as TLS
+tickets may be single-use.
+
+When providing the SNI value to the application, TLS 1.3 requires the value
+from the most recent ClientHello to be used ({{RFC8446, Section 4.6.1}}).  If the
+server TLS implementation violates that requirement and instead uses the legacy
+behavior of reporting the SNI value of the original session, this can lead to a
+confusion attack where the client and the server disagree on the server name
+being used (similar to the attacks described in {{DB15}}).  It is thus crucial
+for implementers to ensure that this aspect of SNI processing is handled
+correctly before enabling cross-name resumption.
```

Comment:
"It is thus crucial" might be better replaced with a MUST, lest we mint another RFC 6919 phrase.

Reply:
That's fair, though this is largely restating an existing MUST in RFC 8446 with a bit of context about what can go wrong if it's violated in this specific scenario.